src/geometry/CTiglBSplineApproxInterp.cpp

```diff
@@ -78,11 +78,14 @@ namespace tigl
 /**
  * @brief Copies the curve points
  */
-CTiglBSplineApproxInterp::CTiglBSplineApproxInterp(const TColgp_Array1OfPnt &points, int nControlPoints, int degree)
+CTiglBSplineApproxInterp::CTiglBSplineApproxInterp(const TColgp_Array1OfPnt &points, int nControlPoints, int degree, bool continuous_if_closed)
     : m_pnts(1, points.Length())
     , m_indexOfApproximated(static_cast<size_t>(points.Length()))
     , m_degree(degree)
     , m_ncp(nControlPoints)
+    , is_closed(points.Value(points.Lower()).IsEqual(points.Value(points.Upper()),1e-5))
```

Comment:
You should not use a hard-coded threshold! What if the whole curve is tiny? Instead, get the overall size of the curve and use a very small fraction of the overall size. Or, just use a very small threshold for the curve. 1e-5 is definitely too large.

Also, the computation isClosed should be a private method. The reason is, that the point could change theoretically. Then you always have to update your states, which is not good practice.